simple-acquaintances/data/datagen.py

```diff
@@ -0,0 +1,400 @@
+            if self.outputDir[0] == '~':
+                prefix = os.path.expanduser(self.outputDir)
+            else:
+                prefix = self.outputDir
+        else:
+            prefix = ''
+
+        for key in data:
+            # We have the same keys for data and its corresponding path.
+            with open(os.path.join(prefix, paths[key]), 'w') as f:
+                if key == 'config':
+                    json.dump(data[key], f, indent = 4, sort_keys = True)
+                else:
+                    for dataItem in data[key]:
+                        f.write("\t".join([str(d) for d in dataItem]) + "\n")
+
+    def __prepareDataToWrite(self, knowsMatrix, people):
+        """
+        Prepare data for writing to file.
+        """
+        n = knowsMatrix.shape[1]
+
+        knowsData = []
+        knowsObs = []
+        knowsTarget = []
+        knowsTruth = []
+        
+        totalCombinations = knowsMatrix.shape[0] * (knowsMatrix.shape[1] - 1)
+        targets = np.random.choice(range(totalCombinations), 
+                                    int(self.targetSplit * totalCombinations), replace = False)
+        for i in range(knowsMatrix.shape[0]):
+            for j in range(knowsMatrix.shape[1]):
+                if i != j:
+                    knowsData.append([i, j, knowsMatrix[i][j]])
+                if (n * i + j) not in targets and i != j:
+                    knowsObs.append([i, j, knowsMatrix[i][j]])
+                elif (n * i + j) in targets and i != j:
+                    knowsTarget.append([i, j])
+                if (n * i + j) in targets and i != j:
+                    knowsTruth.append([i, j, knowsMatrix[i][j]])
+
+        livedObs = []
+        for person in people:
+            for index, place in enumerate(person.lived):
+                if place == 1:
+                    livedObs.append([person.index, index])
+
+        likesObs = []
+        for person in people:
+            for globalLike in person.globalLikes:
+                likesObs.append([person.index, globalLike, person.globalLikes[globalLike]])
+
+        #Both global and local likes are combined in a single list called likesObs.
+        for person in people:
+            for localLike in person.localLikes:
+                likesObs.append([person.index, len(self.globalThings) + localLike, person.localLikes[localLike]])
+        
+        config = {  'people': self.numberOfPeople,
+                    'places': len(self.places),
+                    'globalThings': len(self.globalThings),
+                    'localThings': len(self.localThings),
+                    'placesLivedMean': self.placesLivedMean,
+                    'PlacesLivedStandardDeviation': self.placesLivedSD,
+                    'localLikesVariance': self.localLikesVariance,
+                    'targetSplit': self.targetSplit,
+                    'randomNumberGeneratorSeed': self.randomSeed,
+                    'outputDirectory': self.outputDir}
```

Comment:
End the brace on the next line at the same indentation level as the start of the statement (the "c" in "config", in this case).

Reply:
Noted.